### PR TITLE
feat!: update to angular 18

### PR DESCRIPTION
**Pull request by gfellerph**
This upgrades the following packages to Angular 18:
- '@swisspost/design-system-intranet-header': major
- '@swisspost/design-system-components-angular': major
- '@swisspost/design-system-styles': major
- '@swisspost/design-system-demo': major

- [x] Update compat table in the docs

### CODE REVIEW

File: packages/components-angular/package.json [hunk 1]
```diff
@@ -10,28 +10,28 @@
...
     "tslib": "2.6.3",
     "zone.js": "0.14.7"
   },
   "devDependencies": {
-    "@angular-devkit/build-angular": "17.3.8",
+    "@angular-devkit/build-angular": "18.0.7",
     "@angular-eslint/builder": "17.5.2",
     "@angular-eslint/eslint-plugin": "17.5.2",
     "@angular-eslint/eslint-plugin-template": "17.5.2",
     "@angular-eslint/template-parser": "17.5.2",
```
Comment (oliverschuerch) [inline packages/components-angular/package.json:32]: Is there a reason, why you didn't upddate those deps?
They're all eslinlt ^8.57.0 compatible.

File: packages/demo/package.json [hunk 2]
```diff
@@ -39,15 +39,15 @@
     "zone.js": "0.14.7"
   },
   "devDependencies": {
-    "@angular-devkit/build-angular": "17.3.8",
+    "@angular-devkit/build-angular": "18.0.7",
     "@angular-eslint/builder": "17.5.2",
     "@angular-eslint/eslint-plugin": "17.5.2",
     "@angular-eslint/eslint-plugin-template": "17.5.2",
     "@angular-eslint/template-parser": "17.5.2",
```
Comment (oliverschuerch) [inline packages/demo/package.json:46]: Same here.

File: packages/documentation/src/stories/getting-started/migration-guide/migrationv7-8.component.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+              will be detailed in the migration guide for v9.
+            </p>
+            <p>
+              Any ng-bootstrap components will be replaced by web components and available for
+              Angular users with the
+              <a
+                href="https://design-system.post.ch/?path=/docs/833ef689-a573-40f5-a6a6-30a999b94733--docs"
+                >@swisspost/design-system-components-angular</a
+              >
+              pacakge. With this change, we will limit our dependency on Angular to the
```
Comment (oliverschuerch) [inline packages/documentation/src/stories/getting-started/migration-guide/migrationv7-8.component.ts:101]:
```suggestion
              package. With this change, we will limit our dependency on Angular to the
```

File: packages/intranet-header-workspace/package.json [hunk 4]
```diff
@@ -28,13 +28,13 @@
     "zone.js": "0.14.7"
   },
   "devDependencies": {
-    "@angular-devkit/build-angular": "17.3.8",
+    "@angular-devkit/build-angular": "18.0.7",
     "@angular-eslint/builder": "17.5.2",
     "@angular-eslint/eslint-plugin": "17.5.2",
     "@angular-eslint/eslint-plugin-template": "17.5.2",
     "@angular-eslint/template-parser": "17.5.2",
```
Comment (oliverschuerch) [inline packages/intranet-header-workspace/package.json:35]: And same here.

File: packages/styles/src/components/pagination.scss [hunk 5]
```diff
@@ -40,6 +40,7 @@
       &[aria-label='Last'] {
         > span {
           display: inline-block;
+          color: transparent;
```
Comment (oliverschuerch) [inline packages/styles/src/components/pagination.scss:43]: Why?

**Review by oliverschuerch — CHANGES_REQUESTED** [review]

File: packages/styles/src/components/pagination.scss (hunk 5 above)
Comment (gfellerph) [reply packages/styles/src/components/pagination.scss:43]: There are default ">>" arrows in the field from ng-bootstrap and they overlap with our icon.

File: packages/components-angular/package.json (hunk 1 above)
Comment (gfellerph) [reply packages/components-angular/package.json:32]: Done

**Review by oliverschuerch — APPROVED** [review]
Do you update the Compatibility Table in the docs in a other PR?

**Review by oliverschuerch — APPROVED** [review]

File: packages/documentation/src/stories/getting-started/migration-guide/migrationv7-8.component.ts [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+                  <code languages="['bash']">npm install @swisspost/design-system-styles@8</code>
+                </p>
+              </li>
+            </ol>
+          </li>
+          <li>
+            <h3>Deprecation notice</h3>
+            <p>
+              This is the last major release that supports
+              <a href="https://getbootstrap.com/">Bootstrap</a> and
```
Comment (alizedebray) [inline packages/documentation/src/stories/getting-started/migration-guide/migrationv7-8.component.ts:78]:
```suggestion
              <a href="https://getbootstrap.com/">bootstrap</a> ${this.angular
                ? html`and
```

File: packages/documentation/src/stories/getting-started/migration-guide/migrationv7-8.component.ts [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+              <a href="https://ng-bootstrap.github.io/#/home">@ng-bootstrap/ng-bootstrap</a>.
```
Comment (alizedebray) [inline packages/documentation/src/stories/getting-started/migration-guide/migrationv7-8.component.ts:79]:
```suggestion
              <a href="https://ng-bootstrap.github.io/#/home">@ng-bootstrap/ng-bootstrap</a>` : nothing}.
```

File: packages/documentation/src/stories/getting-started/migration-guide/migrationv7-8.component.ts [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+              the future, the component structure and class names will remain the same, only the
+              styles in the background will be switched to the new token system.
+            </p>
+            <p>
+              Certain utility classes will no longer be avaiable from the Design System due to this
+              change. However, projects that need them can add the bootstrap library as their
+              dependency and conditionally import the needed files. More information on this change
+              will be detailed in the migration guide for v9.
+            </p>
+            <p>
```
Comment (alizedebray) [inline packages/documentation/src/stories/getting-started/migration-guide/migrationv7-8.component.ts:94]:
```suggestion
            ${this.angular ? html`<p>
```

File: packages/documentation/src/stories/getting-started/migration-guide/migrationv7-8.component.ts [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+              Any ng-bootstrap components will be replaced by web components and available for
+              Angular users with the
+              <a
+                href="https://design-system.post.ch/?path=/docs/833ef689-a573-40f5-a6a6-30a999b94733--docs"
+                >@swisspost/design-system-components-angular</a
+              >
+              package. With this change, we will limit our dependency on Angular to the
+              components-angular package which allows us to ship Angular upgrades much faster in the
+              future.
+            </p>
```
Comment (alizedebray) [inline packages/documentation/src/stories/getting-started/migration-guide/migrationv7-8.component.ts:104]:
```suggestion
            </p>` : nothing}
```

**Review by alizedebray — APPROVED** [review]

**Pull request merged**